### PR TITLE
CHI-988: Enhanced case route tests

### PR DESCRIPTION
**Pull request by stephenhand**
Primary Reviewer: @murilovmachado 

## Description

* Extra test cases for the case PUT endpoint
* Extra test cases for the case list endpoint
* Added extra validation to existing tests, to validate DB state, as well as response content for write endpoints

### Checklist
- [ X ] Corresponding issue has been opened
- [ X ] New tests added
- [ N/A ] Strings are localized

### Related Issues
Fixes CHI-988

### Verification steps

Tests run on PRs

### CODE REVIEW

File: tests/routes/case.test.js [hunk 1]
```diff
@@ -137,6 +168,122 @@ describe('/cases route', () => {
...
+        },
+        {
+          description: 'should return first X cases when limit X is specified',
+          listRoute: `/v0/accounts/${accounts[0]}/cases?limit=3`,
+          expectedCasesAndContacts: () =>
+            createdCasesAndContacts
+              .filter(ccc => ccc.case.dataValues.accountSid === accounts[0])
+              .sort((ccc1, ccc2) => ccc2.case.id - ccc1.case.id)
+              .slice(0, 3),
+          expectedCount: 5,
```
Comment (murilovmachado) [inline tests/routes/case.test.js:243]: I'm a litle confused with `expectedCount`, shouldn't it be `3`?

File: tests/routes/case.test.js [hunk 2]
```diff
@@ -137,6 +168,122 @@ describe('/cases route', () => {
...
+        {
+          description:
+            'should return X cases, starting at Y when limit X and offset Y are specified',
+          listRoute: `/v0/accounts/${accounts[0]}/cases?limit=2&offset=1`,
+          expectedCasesAndContacts: () =>
+            createdCasesAndContacts
+              .filter(ccc => ccc.case.dataValues.accountSid === accounts[0])
+              .sort((ccc1, ccc2) => ccc2.case.id - ccc1.case.id)
+              .slice(1, 3),
+          expectedCount: 5,
```
Comment (murilovmachado) [inline tests/routes/case.test.js:254]: same question on `expectedCount` here

File: tests/routes/case.test.js [hunk 3]
```diff
@@ -137,6 +168,122 @@ describe('/cases route', () => {
...
+        {
+          description:
+            'should return remaining cases, starting at Y when limit X and offset Y are specified',
+          listRoute: `/v0/accounts/${accounts[0]}/cases?offset=2`,
+          expectedCasesAndContacts: () =>
+            createdCasesAndContacts
+              .filter(ccc => ccc.case.dataValues.accountSid === accounts[0])
+              .sort((ccc1, ccc2) => ccc2.case.id - ccc1.case.id)
+              .slice(2),
+          expectedCount: 5,
```
Comment (murilovmachado) [inline tests/routes/case.test.js:265]: Here de description says `limit` and `offset` are specified, but looking at the URL we're specifying only the `offset`

**Review by murilovmachado — APPROVED** [review]
Very nice! I didn't know about `jest-each`, it seems very useful for our scenarios.
I had a few questions, but overall it's great!

File: tests/routes/case.test.js (hunk 1 above)
Comment (stephenhand) [reply tests/routes/case.test.js:243]: The `expectedCount` refers to the `totalCount` for the whole set of valid records before LIMIT & OFFSET are applied, I will rename it to make that clear

Comment (murilovmachado) [reply tests/routes/case.test.js]: oh, ok! Thanks for renaming it, much clearer now

**Pull request merged**